js/app/application.js

```diff
@@ -352,6 +355,15 @@ const Application = new Knowledge.Class({
     },
 
     vfunc_shutdown: function () {
+        // Record a content access metric before shutting down
```

Comment:
It would be better not to introduce all this coupling into application.js, though I'm not sure if it's really possible. Is there any way we could keep this inside HistoryStore?